File: Server/GameServer/codeGen2/testcode_templates/cpp/HandleMethod.cpp

INT /*::PROTOCLASSNAME::*/::_/*::HANDLER_METHOD_NAME::*/(const char *__src,INT    __len)
{
		/*::XPACKET_CLASS_NAME::*/Ptr pPacket;
		pPacket.CreateInstance();
		if(pPacket==NULL) return -1;
		if(XFROM_BUFFER(pPacket,__src,__len)!=__len)
		{
			return -1;
		}
		int __xvres = /*::HANDLER_METHOD_NAME::*/(/*::HANDLER_PASS_PARAMETERS::*/);
		return __xvres;
}

File: Server/GameServer/codeGen2/testcode_templates/php/HandleMethod.php

		private function _/*::HANDLER_METHOD_NAME::*/($__src)
		{
			//require_once("/*::XPACKET_CLASS_NAME::*/.php");
			$pPacket  =  new /*::XPACKET_CLASS_NAME::*/();
			
			if(!$this->useAMF)
			{
				$__len		= $__src->getBytesAvailable();
				if($pPacket->FromBuffer($__src)!=$__len)
				{
					return -1;
				}
			
			//	if (DEBUG)
			//	{
			//	    Logger::getLogger()->debug("/*::HANDLER_METHOD_NAME::*/".$pPacket->ToDebugString());
			//	}
				
				return $this->/*::HANDLER_METHOD_NAME::*/($pPacket);			
			}
			else
			{
				

				$obj = ReadAMF3Object($__src);
				$pPacket->fromAMFObject($obj);
				
			//	if (DEBUG)
			//	{
			//	    Logger::getLogger()->debug("/*::HANDLER_METHOD_NAME::*/".$pPacket->ToDebugString());
			//	}

				return $this->/*::HANDLER_METHOD_NAME::*/($pPacket);
			}


		}